src/java/org/apache/cassandra/index/sai/disk/v3/CassandraDiskAnn.java

```diff
@@ -175,20 +138,20 @@ private SearchResult searchInternal(GraphSearcher<float[]> searcher, float[] que
             scoreFunction = compressedVectors.approximateScoreFunctionFor(queryVector, similarityFunction);
             reranker = i -> similarityFunction.compare(queryVector, threadLocalView.getVector(i));
         }
-        var result = searcher.search(scoreFunction,
-                                     reranker,
-                                     topK,
-                                     threshold,
-                                     ordinalsMap.ignoringDeleted(acceptBits));
-        context.addAnnNodesVisited(result.getVisitedCount());
+        var searcher = new GraphSearcher.Builder<>(threadLocalView).build();
+        var result = searcher.search(scoreFunction, reranker, topK, threshold, ordinalsMap.ignoringDeleted(acceptBits));
         Tracing.trace("DiskANN search visited {} nodes to return {} results", result.getVisitedCount(), result.getNodes().length);
-        return result;
+        // Threshold based searches do not support resuming the search.
```

Comment:
Untested but "it should work", thing is you'd have to pass a different threshold since it's designed to find ~all the results in a single call.  So you could call resume with `lastThreshold * 0.9`

Reply:
LMK if you want to try this and I will add a test to jvector to get more confidence on it

Reply:
Our only threshold queries are geo distance, and those don't have a practical limit in the vector index, so I don't think we need that yet.

Reply:
Can we change the comment to "... do not need to resume the search"?

Reply:
I updated the comment.